src/Polly.RateLimiting/RateLimiterResilienceStrategy.cs

```diff
@@ -65,7 +65,11 @@ protected override async ValueTask<Outcome<TResult>> ExecuteCore<TResult, TState
             await OnLeaseRejected(new OnRateLimiterRejectedArguments(context, lease)).ConfigureAwait(context.ContinueOnCapturedContext);
         }
 
-        var exception = retryAfter.HasValue ? new RateLimiterRejectedException(retryAfter.Value) : new RateLimiterRejectedException();
+        var exception = retryAfter.HasValue
+            ? new RateLimiterRejectedException(retryAfter.Value)
+            : new RateLimiterRejectedException();
```

Comment:
Then we don't pay (IIRC) pay the penalty of `.Value` checking `HasValue` internally?

```suggestion
        var exception = retryAfter is { } retryAfterValue
            ? new RateLimiterRejectedException(retryAfterValue)
            : new RateLimiterRejectedException();
```